test/builder_quantization.cpp

```diff
@@ -1182,6 +1184,31 @@ TEST(builder, scaled_quantize_concat_unsigned_varying)
               read_vector<uint8_t>(result));
 }
 
+TEST(builder, scaled_QDotInteger)
+{
+    Shape shape_a{3, 3}; // input shape
+    vector<uint8_t> a_data = {1, 2, 2, 3, 4, 5, 1, 0, 1};
+    Shape shape_b{3, 3}; // filter shape
```

Comment:
Change to {2, 4} or something where the dim sizes are not equal. This will help with whether QD is [m,k] * [n,k] or [m,k] * [k,n]